### PR TITLE
add entry for the standard library preview repository

### PR DESCRIPTION
**Pull request by glessard**
### Pull Request Description

Add an entry for the standard library preview repository.
On Linux, the project requires the --enable-test-discovery flag to be passed in order to build on Swift 5.1 through 5.3.
I don't know where to add this.

This project supports Swift 5.1 and up; there is no Swift 4.0 branch.

### Acceptance Criteria

To be accepted into the Swift source compatibility test suite, a project must:

- [x] be an *Xcode* or *swift package manager* project
- [x] support building on either Linux or macOS
- [x] target Linux, macOS, or iOS/tvOS/watchOS device
- [x] be contained in a publicly accessible git repository
- [ ] maintain a project branch that builds against Swift 4.0 and passes any unit tests
- [x] have maintainers who will commit to resolve issues in a timely manner
- [x] be compatible with the latest GM/Beta versions of *Xcode* and *swiftpm*
- [x] add value not already included in the suite
- [x] be licensed with one of the following permissive licenses:
	* BSD
	* MIT
	* Apache License, version 2.0
	* Eclipse Public License
	* Mozilla Public License (MPL) 1.1
	* MPL 2.0
	* CDDL
- [x] pass `./project_precommit_check` script run

Ensure project meets all listed requirements before submitting a pull request.

### CODE REVIEW

File: projects.json [hunk 1]
```diff
@@ -2778,6 +2778,32 @@
...
   },
+  {
+    "repository": "Git",
+    "url": "https://github.com/apple/swift-standard-library-preview.git",
+    "path": "swift-standard-library-preview",
+    "branch": "main",
+    "maintainer": "guillaume.lessard@apple.com",
+    "compatibility": [
+      {
+        "version": "5.1",
```
Comment (shahmishal) [inline projects.json:2789]: Can we use swift 5.0 here? We would like to only list swift programming lang (4.0, 4.2, and 5.0). 
```suggestion
        "version": "5.0",
```

**Comment by shahmishal** [comment]
@swift-ci test

File: projects.json (hunk 1 above)
Comment (glessard) [reply projects.json:2789]: The package manifest has a tools-version of 5.1. It will certainly not build on 5.0 on Linux.

Comment (shahmishal) [reply projects.json:2789]: the version info is used for swift flag `--swift-version`. Also, we will be building the source compat suite with main and release (5.5) branch.

**Comment by glessard** [comment]
Is there a way to add a flag for the `swift test` invocation? On Linux for 5.1, 5.2 and 5.3, we would need to pass `--enable-test-discovery`.

**Comment by shahmishal** [comment]
> Is there a way to add a flag for the swift test invocation? On Linux for 5.1, 5.2 and 5.3, we would need to pass --enable-test-discovery.
We currently don't support testing on source compat suite.

**Comment by glessard** [comment]
The flag is required for `swift build` as well. It looks quite weird there.

**Comment by glessard** [comment]
@swift-ci test

**Comment by glessard** [comment]
It won't build on main because SwiftPM made a source-incompatible change.
See https://bugs.swift.org/browse/SR-15558

**Comment by glessard** [comment]
@shahmishal This should now build on main, as the source-incompatible change in SwiftPM was fixed in https://github.com/apple/swift-package-manager/pull/3922.

**Comment by glessard** [comment]
@swift-ci test

**Review by clackary — APPROVED** [review]
LGTM!

**Pull request merged**